### ISSUE TITLE
MessageBus.callbackInterval does not work in production

### ISSUE THREAD

**Issue by ChristofferJoergensen** [SamSaffron/message_bus#129]
Thanks for making this project available. I've been using it for a few months now without altering the default `MessageBus.callbackInterval`. Now I need to increase the frequency, which I've done successfully in my development env. However, when deploying to production it seems that the second line below is ignored:

```
MessageBus.start();
MessageBus.callbackInterval = 1800;
MessageBus.subscribe("/some_id", function(status) {
  // work
});
```

Any idea how I could debug this?

**Comment by SamSaffron** [comment]
hmm why are you increasing the interval, the long polling gives you complete coverage anyway. 

as an aside this should absolutely work per:

https://github.com/SamSaffron/message_bus/blob/master/assets/message-bus.js#L241-L248

But keep in mind there is a backgroundCallbackInterval you may also want to tweak.

**Comment by ChristofferJoergensen** [comment]
So the `callbackInterval` has no effect on how fast new published data gets picked up by the web interface? The reason I asked in the first place is because I am experiencing that it could take up to 30 seconds for the web interface to pick up changes.

**Comment by SamSaffron** [comment]
no it should be delivered right away, see https://chat.samsaffron.com/ for
it in action, it runs default settings, try 2 web browsers.

**Comment by ChristofferJoergensen** [comment]
Hm, strange. Do you have a suggestion on how to debug the delay? I am publishing from a background job (Sidekiq to be specific) btw, not from a controller or another web page.

**Comment by SamSaffron** [comment]
Are you behind a proxy or something that is killing long polling? First
thing I would do is introduce HTTPS

**Comment by danweller18** [comment]
@ChristofferJoergensen you can use chrome to view the XHR request and that will show you every time the web page is polling.... based on your settings it should show a request every 1.8 seconds. Go to the network tab -> Start recording log -> click on XHR tab -> the name of each request should be something like `poll?dlp=t`

Not sure if this will help but I also had problems with asset pipeline at first and had to set the callbackInterval outside of the document.ready function- in order for rails to pick up my custom callbackInterval.

```javascript
$(document).ready(function () {
  MessageBus.subscribe('/channel', function(msg) {
//do your stuff
  });
});

MessageBus.enableLongPolling = false;
MessageBus.callbackInterval = 1500;
```

**Comment by SamSaffron** [comment]
note callbackInterval of 1500 is super duper aggressive and not something I
would ever recommend unless you are on an HTTP 1.0 only network.

**Comment by ChristofferJoergensen** [comment]
If the standard `callbackInterval` is giving full coverage I don't need to increase it. So the remaining question would be why feedback isn't instant. It seems that we had forgotten to add the following to the nginx config:

    location /message-bus/ {
      proxy_buffering off;
      proxy_http_version 1.1;
    }

However, putting it in made our polling requests return a 404. So we had to remove it from the config again.

Btw, we are using https.

I will investigate further now.

**Comment by SamSaffron** [comment]
Inheritance in NGINX is weird, you got to add a directive there explaining
where it can get stuff for message bus from.

**Comment by ChristofferJoergensen** [comment]
I've tried to fix the problem without luck. While testing I've noticed something:

I have 2 tabs open, and in the first tab I push a button that, on the server, triggers a background job that a few seconds later finishes with a MessageBus callback to the browser.

The first tab where I pushed the button gets the callback after several seconds.
The other tab gets the callback immediately.

Could this discovery be the key to understanding what's wrong?
